cmd/main.go

```diff
@@ -17,6 +17,7 @@ limitations under the License.
 package main
 
 import (
+	"context"
```

Comment:
This file is very hacky at the moment. Just to test things out. It ill need to be reworked still.

The other files should be in pretty good shape.